src/ncp/ncp_uart.cpp

```diff
@@ -176,7 +176,12 @@ void NcpUart::EncodeAndSendToUart(void)
 
     if (len > 0)
     {
-        otPlatUartSend(mUartBuffer.GetBuffer(), len);
+        if (otPlatUartSend(mUartBuffer.GetBuffer(), len) != kThreadError_None)
```

Comment:
I was looking at implementations of `otPlatUartSend` and as far as I can tell, all of them only fail if they are busy handling the previous send, and return `kThreadError_Busy`. Practically, if that happens, that means some how `NcpUart` called `otPlatUartSend` again, before it got a call back to `otPlatUartSendDone`, right? And that shouldn't ever happen.

But ignoring that, generally I agree this is a better design for `NcpUart`, but I was thinking if `NcpUart` could handle the `kThreadError_Busy` specifically. For instance, if it has the buffer, should it just queue it up and wait (for what?) and try again?